projects/client/RabbitMQ.Client/src/client/api/AmqpTcpEndpoint.cs

```diff
@@ -138,6 +140,11 @@ public AmqpTcpEndpoint CloneWithHostname(string hostname)
             return new AmqpTcpEndpoint(hostname, _port, Ssl);
         }
 
+        /// <summary>
+        /// The default Amqp SSL protocols.
+        /// </summary>
+        public static SslProtocols DefaultAmqpSslProtocols { get; set; } = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
```

Comment:
Does it make sense for this field to be an `AmqpTcpEndpoint` field and not, say, `ConnectionFactory`?

Reply:
I agree it would make sense to keep global config in the connection factory.

Reply:
That was my first thought. But than I saw a `DefaultAmqpSslPort` in `AmqpTcpEndpoint`...

For me, it makes a lot more sense to honor the SSL options already part of `ConnectionFactory` when using an URI than having `AmqpTcpEndpoint` run free with a mind if its own.

Whst do you think?

Reply:
I'd move both this field and `DefaultAmqpSslPort ` to connection factory and bump the version to `5.2.0`. @kjnilsson WDYT?

Reply:
Moving `DefaultAmqpSslPort` to a different class is a breaking change, so a new major is needed if you do that.

Reply:
It's a rarely used directly field. It would be a true facepalm to release a 6.0 because of that.

Reply:
Well, this new field can go into `ConnectionFactory` for 5.x and we can move the port one in master only.

Reply:
We can leave DefaultAmqpSslPort, mark it as obsolete and move it to ConnectionFactory.

We are gradually accumulating breaking and proposed breaking changes to warrant considering a major release at some point in the not so distant future.

Reply:
That would keep it binarily compatible while directing development to the new way.

Reply:
Sounds like we have a consensus then? Note that the port property move ideally should be in a separate PR.

Reply:
Property or constant filed?

And the `DefaultAmqpSslProtocols` would still be a static property, right? And with the good defaults, right?

Reply:
So, what's the verdict?

Reply:
@paulomorgado in this PR, let's move the field to `ConnectionFactory`. In a separate one for master, move the port one with a backwards compatible reference. Both @kjnilsson and I are leaning towards using properties for no particular reason, merely as a matter of taste.

Thank you!

Reply:
It's not a matter of taste. It's semantic.

I've split SSL protocols settings into static and instance properties and named them to convey the meaning that they will only be used with URIs.

As for the default port, that one can be overridden in the URI and, isn't 5671 the default as 80 is for HTTTP and 443 for HTTPS? If it is, it's pretty fine where it is. `System.Uri` has default ports per syntax.